docs: release notes for v2.27 (#28121)
gatsbyjs/gatsby: docs/reference/release-notes/v2.27/index.md

---
date: "2020-11-19"
version: "2.27.0"
---

# [v2.27](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.27.0-next.0...gatsby@2.27.0) (November 2020 #2)

---

Welcome to `gatsby@2.27.0` release (November 2020 #2).

Key highlights of this release:

> - [Create-gatsby](#create-gatsby) - new, interactive way to create a Gatsby site

Reviewer: ```suggestion
- [create-gatsby](#create-gatsby) - new, interactive way to create a Gatsby site
```

- [Experimental: Queries on Demand](#experimental-queries-on-demand) - improves `gatsby develop` bootup time
- [Experimental: Lazy page bundling](#experimental-lazy-page-bundling-in-development) - also improves `gatsby develop` bootup time
- [Experimental: SSR in develop](#experimental-ssr-in-development) - catch SSR errors early
- [gatsby-plugin-mdx@1.5.0](#gatsby-plugin-mdx150) - new option for better performance
- [Notable bugfixes](#notable-bugfixes)

Sneak peek to next releases:

- [Experimental: Lazy images in develop](#experimental-lazy-images-in-develop)
- [Documentation Reorganization](#documentation-reorganization)

**Bleeding Edge:** Want to try new features as soon as possible? Install `gatsby@next` and let us know
if you have any [issues](https://github.com/gatsbyjs/gatsby/issues).

[Previous release notes](../v2.26/index.md)<br>
[Full changelog](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.27.0-next.0...gatsby@2.27.0)

> ### Create-gatsby

Reviewer: ```suggestion
### create-gatsby
```

There is now a new, interactive way to create a Gatsby site. As long as you have Node.js installed you're ready to go.

> ```

Reviewer: ```suggestion
```shell
```

npm init gatsby
```

> This command will trigger a series of prompts, allowing you to select your CMS, preferred styling tools and common Gatsby plugins. After you've made your selections let the installation complete and you'll have a working Gatsby site, all packages included and configured for use.

Reviewer 2: ```suggestion
This command will ask a series of questions, allowing you to select your CMS, preferred styling tools and common Gatsby plugins. After you've made your selections let the installation complete and you'll have a working Gatsby site, all packages included and configured for use.
```

### Making `gatsby develop` faster

The Gatsby develop server can get slow to start on larger sites. We're working hard to speed it up. We're addressing different scaling problems one by one and have shipped several improvements behind flags as detailed below. If you'd like to enable all these dev speedups (along with all future ones!), we've created a single flag for you. Run your site like `GATSBY_EXPERIMENTAL_FAST_DEV=true gatsby develop` and zoom zoom!

### Experimental: Queries on Demand

When developing a Gatsby site there's no need to run all graphql queries before serving the site.
Instead, Gatsby will run queries for pages as they're requested by a browser.
Think of it like lazily loading the data your pages need, when they need it!

This avoids having to wait for slower queries (like image processing) if you're editing an unrelated part of a site.
What this means for you: faster local development experience, up to 2x faster in many cases!

This feature is available under a flag.

> ```sh
> cross-env GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND=1 gatsby develop

Reviewer: ```suggestion
```shell
GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND=1 gatsby develop
```

We don't use cross-env on the other prompts

```

Please try it and [let us know](https://github.com/gatsbyjs/gatsby/discussions/27620) if you have any issues
(it may become a default for `develop` in the future)

[Details and discussion](https://github.com/gatsbyjs/gatsby/discussions/27620).

## Experimental: SSR in Development

> One of the least enjoyable bugs to encounter in Gatsby is when your build fails due to code trying to reference `window` or `document` or other browser globals that are not accessible in node.js during SSR.

Reviewer: ```suggestion
One of the least enjoyable bugs to encounter in Gatsby is when your build fails due to code trying to reference `window` or `document` or other browser globals that are not accessible in Node.js during SSR.
```

---
date: "2020-11-19"
version: "2.27.0"
---

# [v2.27](https://github.com/gatsbyjs/gatsby/compare/gatsby@2.27.0-next.0...gatsby@2.27.0) (November 2020 #2)

---

Welcome to `gatsby@2.27.0` release (November 2020 #2).

Key highlights of this release:

- [Queries on Demand](#queries-on-demand) - improves `gatsby develop` bootup time
- [Add your item here](#link-to-header) - one-line summary

Other notable changes and bugfixes:

- [Add your item here](#link-to-header) - one-line summary

Sneak peek to next releases:

- [Experimental: Lazy images in develop](#experimental-lazy-images-in-develop)
- [Documentation Reorganization](#documentation-reorganization)

**Bleeding Edge:** Want to try new features as soon as possible? Install `gatsby@next` and let us know
if you have any [issues](https://github.com/gatsbyjs/gatsby/issues).

[Previous release notes](../v2.26/index.md)

### Queries on Demand

When developing a Gatsby site there's no need to run all graphql queries before serving the site.

> Instead, Gatsby could run queries for pages as they're requested by a browser.

Reviewer 3: ```suggestion
Instead, Gatsby will run queries for pages as they're requested by a browser. Think of it like lazily loading the data your pages need, when they need it!
```

> This would avoid having to wait for slower queries (like image processing) if you're editing an unrelated part of a site.

Reviewer 3: ```suggestion
This avoids having to wait for slower queries (like image processing) if you're editing an unrelated part of a site. What this means for you: faster local development experience, up to 2x faster in many cases!
```

This feature is available under a flag.

```sh
cross-env GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND=1 gatsby develop
```

Please try it and [let us know](https://github.com/gatsbyjs/gatsby/discussions/27620) if you have any issues
(it may become a default for `develop` in the future)

[Details and discussion](https://github.com/gatsbyjs/gatsby/discussions/27620).

## Work in progress

### Experimental: Lazy images in develop

> We've got some feedback that the more image-heavy your website gets, the slower `gatsby develop`.

Reviewer 3: ```suggestion
We've got some feedback that the more images your website contains, the slower your local development experience gets. You spend time waiting for images to process, instead of you know, developing! No longer!
```

Reviewer 3: cc @hashimwarren feel free to add some _pizzazz_ 👍
